### ISSUE TITLE
Bug: ("export * as" reports Parsing error: Unexpected token as)

### ISSUE THREAD

**Issue by rickmed** [eslint/eslint#16827]
### Environment

Node version: 18.13.0
npm version: 8.19.3
Local ESLint version: 8.32.0
Global ESLint version: n/a
Operating System: windows with wsl2

### What parser are you using?

Default (Espree)

### What did you do?

<details>
<summary>Configuration</summary>

.eslintrc
```
{
  "extends": ["eslint:recommended"],
  "env": {
    "es2016": true
  },
  "parserOptions": {
    "sourceType": "module"
  },
  "rules": {
    "semi": ["error", "never"],
    "quotes": ["error", "double", { "allowTemplateLiterals": true }]
  }
}

```
</details>

```js
export * as check from "./source/check.js"
```

### What did you expect to happen?

no errors

### What actually happened?

got:
```Parsing error: Unexpected token as```

### Participation

- [ ] I am willing to submit a pull request for this issue.

### Additional comments

the code works fine

**Comment by snitin315** [comment]
Hi @rickmed, The `export * as foo from bar` syntax was introduced [in ECMAScript 2020. ](https://eslint.org/blog/2020/06/eslint-v7.2.0-released/#highlights)

From your configuration, you seem to be using ES2016, Please [update your configuration to use 2020](https://eslint.org/docs/latest/use/configure/language-options#specifying-parser-options) or later:

```diff
{
  "extends": ["eslint:recommended"],
  "env": {
    "es2016": true
  },
  "parserOptions": {
    "sourceType": "module",
+   "ecmaVersion": 2020,
  },
  "rules": {
    "semi": ["error", "never"],
    "quotes": ["error", "double", { "allowTemplateLiterals": true }]
  }
}
```
